libavformat/movenccenc.c

```diff
@@ -640,5 +640,12 @@ void ff_mov_cenc_free(MOVMuxCencContext* ctx)
     if (ctx->encryption_scheme == MOV_ENC_CENC_AES_CBC_PATTERN && ctx->aes_cbc) {
         av_freep(&ctx->aes_cbc);
     }
-    av_freep(&ctx->auxiliary_info);
+    if (ctx->auxiliary_info_size > 0) {
```

Comment:
- What is the reason to check auxiliary_info_size?
- Shouldn't the condition be:
```
    if (ctx->auxiliary_info_alloc_size > 0) {
      ...
    }
```
- I don't think we need to check any condition though. Can you add a comment and explain the reason if it is needed.

Reply:
Changed to `auxiliary_info_alloc_size`

The check is needed because memory may not be allocated (the init right above does not do it)

Reply:
If memory is not allocated ctx->auxiliary_info should be NULL and freeing a NULL pointer is OK.

Reply:
Ah I see. Yeah the check isn't needed then